pytransform3d/transform_manager/_temporal_transform_manager.py

```diff
@@ -0,0 +1,86 @@
+import abc
+from typing import Any
+
+import numpy as np
+
+from ._transform_graph_base import TransformGraphBase
+from ..transformations import (check_transform, invert_transform, concat)
+
+
+class TimeTransform(abc.ABC):
+    """Base class passed to TemporalTransformManager.add_transform() to deal with time.
+    """
+    
+    @abc.abstractmethod
```

Comment:
Alternatively we could describe the object passed as a `Protocol`, by that way, clients do not have to subclass `TimeTransform` and can just implement `as_matrix` method ... which would decouple code ...

Reply:
It would decouple code, but errors would still occur only at runtime. At least with this approach you know that you forgot to implement a required function.